spec/filters/truncate_spec.rb

```diff
@@ -76,22 +137,50 @@
     subject { described_class.new("length_bytes" => length, "fields" => [ "example" ]) }
     let(:event) { LogStash::Event.new("example" => list) }
 
-    before { subject.filter(event) }
+    context "stress test" do
+      before { subject.filter(event) }
 
-    stress_it "should truncate all elements in a list" do
-      count.times do |i| 
-        expect(event.get("[example][#{i}]").bytesize).to be <= length
+      stress_it "should truncate all elements in a list" do
+        count.times do |i|
+          expect(event.get("[example][#{i}]").bytesize).to be <= length
+
+          if event.get("[example][#{i}]").bytesize > length
+            expect(subject).to receive(:filter_matched).once
+          else
+            expect(subject).not_to receive(:filter_matched)
+          end
+        end
       end
     end
-  end
 
-  describe "calling of filter_matched" do
-    subject { described_class.new("length_bytes" => 1, "fields" => [ "example" ]) }
-    let(:event) { LogStash::Event.new("example" => 1000) }
+    context "containing elements greater than size" do
+      let(:count) { 10 }
+      let(:list) { count.times.map { Flores::Random.text(100) } }
+      let(:length) { 50 }
 
-    it "should be called" do
-      expect(subject).to receive(:filter_matched).once
-      subject.filter(event)
+      it "should truncate all elements" do
+        expect(subject).to receive(:filter_matched).once
+
+        subject.filter(event)
+
+        count.times do |i|
+          expect(event.get("[example][#{i}]").bytesize).to be <= length
+        end
+      end
+    end
+
+    context "containing elements with mixed sizes" do
+      let(:count) { 10 }
+      let(:list) { (count - 1).times.map { Flores::Random.text(20) } + [Flores::Random.text(100)]}
+      let(:length) { 50 }
+
+      it "should truncate elements that exceed the length" do
+        expect(subject).to receive(:filter_matched).once
+
+        subject.filter(event)
+
+        expect(event.get("[example][#{count - 1}]").bytesize).to be <= length
```

Comment:
should it be in `count.times` loop ?

Reply:
No, because in the setup we have `length = 50` and the `list` is composed of 9 random text width 20 chars plus 1 width 100. We know that only the latest must be truncated, the others are already minor of 50.